pkg/repository/prisma/workflow_run.go

```diff
@@ -725,354 +753,432 @@ func workflowRunMetricsCount(ctx context.Context, pool *pgxpool.Pool, queries *d
+
+				if *opt.ChildIndex < 0 {
+					l.Warn().Msgf("child index must be greater than or equal to 0 but it is : %d", *opt.ChildIndex)
+					// 	return nil, errors.New("child index must be greater than or equal to 0 but it is : " + strconv.Itoa(*opt.ChildIndex))
+				}
+
+				if *opt.ChildIndex < math.MinInt32 || *opt.ChildIndex > math.MaxInt32 {
+					return nil, errors.New("child index must be within the range of a 32-bit signed integer")
+				}
+				createParams.ChildIndex = pgtype.Int4{
+					Int32: int32(*opt.ChildIndex), // nolint: gosec
+					Valid: true,
+				}
+			}
+
+			if opt.ChildKey != nil {
+				createParams.ChildKey = sqlchelpers.TextFromStr(*opt.ChildKey)
+			}
+
+			if opt.ParentId != nil {
+				createParams.ParentId = sqlchelpers.UUIDFromStr(*opt.ParentId)
 			}
-			createParams.Additionalmetadata = additionalMetadataBytes
 
-			// if additional metadata contains a "dedupe" key, use it as the dedupe value
-			if dedupeValue, ok := opts.AdditionalMetadata["dedupe"]; ok {
-				if dedupeStr, ok := dedupeValue.(string); ok {
-					opts.DedupeValue = &dedupeStr
+			if opt.ParentStepRunId != nil {
+				createParams.ParentStepRunId = sqlchelpers.UUIDFromStr(*opt.ParentStepRunId)
+			}
+
+			if opt.AdditionalMetadata != nil {
+				additionalMetadataBytes, err := json.Marshal(opt.AdditionalMetadata)
+				if err != nil {
+					return nil, err
 				}
+				createParams.Additionalmetadata = additionalMetadataBytes
+
+			}
 
-				if dedupeInt, ok := dedupeValue.(int); ok {
-					dedupeStr := fmt.Sprintf("%d", dedupeInt)
-					opts.DedupeValue = &dedupeStr
+			if opt.Priority != nil {
+				createParams.Priority = pgtype.Int4{
+					Int32: *opt.Priority,
+					Valid: true,
 				}
 			}
-		}
 
-		if opts.Priority != nil {
-			createParams.Priority = pgtype.Int4{
-				Int32: *opts.Priority,
-				Valid: true,
+			crp := dbsqlc.CreateWorkflowRunsParams{
+				ID:                 createParams.ID,
+				TenantId:           createParams.Tenantid,
+				WorkflowVersionId:  createParams.Workflowversionid,
+				DisplayName:        createParams.DisplayName,
+				ChildIndex:         createParams.ChildIndex,
+				ChildKey:           createParams.ChildKey,
+				ParentId:           createParams.ParentId,
+				ParentStepRunId:    createParams.ParentStepRunId,
+				AdditionalMetadata: createParams.Additionalmetadata,
+				Priority:           createParams.Priority,
+				Status:             "PENDING",
 			}
-		}
 
-		// create the dedupe value
-		if opts.DedupeValue != nil {
-			_, err = queries.CreateWorkflowRunDedupe(
-				tx1Ctx,
-				tx,
-				dbsqlc.CreateWorkflowRunDedupeParams{
-					Tenantid:          pgTenantId,
-					Workflowversionid: sqlchelpers.UUIDFromStr(opts.WorkflowVersionId),
-					Value:             sqlchelpers.TextFromStr(*opts.DedupeValue),
-					Workflowrunid:     sqlchelpers.UUIDFromStr(workflowRunId),
-				},
+			createRunsParams = append(createRunsParams, crp)
+
+			var desiredWorkerId *string
+
+			if opt.DesiredWorkerId != nil {
```

Comment:
I thought we discussed this casing yesterday, but perhaps I missed something so would be good to get it down in writing -- this isn't the only case where we want to write sticky assignment, I believe to keep the behavior consistent with the current behavior we'll need to write it every single time?

Reply:
ah yes, I thought it was barfing cause we weren't providing a desiredWorkerId, I think the problem was a unique constraint on workflowRunId - should be fixed now